ui/tests/acceptance/jobs-list-test.js

```diff
@@ -735,4 +735,477 @@ module('Acceptance | jobs list', function (hooks) {
+        createJobs(server, 10);
+        await JobsList.visit();
+        assert.dom('.job-row').exists({ count: 10 });
+        let rows = document.querySelectorAll('.job-row');
+        assert.equal(rows.length, 10, 'List is still 10 rows');
+        let modifyIndexes = Array.from(rows).map((row) =>
+          parseInt(row.getAttribute('data-test-modify-index'))
+        );
+        assert.deepEqual(
+          modifyIndexes,
+          Array(10)
+            .fill()
+            .map((_, i) => i + 1)
+            .reverse(),
+          'Jobs are sorted by modify index'
+        );
+        assert.dom('[data-test-pager="next"]').isDisabled();
+
+        // Create a new job
+        server.create('job', {
+          namespaceId: 'default',
+          resourceSpec: Array(1).fill('M: 256, C: 500'),
+          groupAllocCount: 1,
+          modifyIndex: 11,
+          createAllocations: false,
+          shallow: true,
+          name: 'new-job',
+        });
+
+        const controller = this.owner.lookup('controller:jobs.index');
+
+        let currentParams = {
+          per_page: 10,
+        };
+
+        // We have to wait for watchJobIDs to trigger the "dueling query" with watchJobs.
+        // Since we can't await the watchJobs promise, we set a reasonably short timeout
+        // to check the state of the list after the dueling query has completed.
+        await controller.watchJobIDs.perform(currentParams, 0);
+
+        let updatedJob = assert.async(); // watch for this to say "My tests oughta be passing by now"
+        const duelingQueryUpdateTime = 200;
+
+        assert.timeout(500);
+
+        setTimeout(async () => {
+          // Order should now be 11-2
+          rows = document.querySelectorAll('.job-row');
+          modifyIndexes = Array.from(rows).map((row) =>
+            parseInt(row.getAttribute('data-test-modify-index'))
+          );
+          assert.deepEqual(
+            modifyIndexes,
+            Array(10)
+              .fill()
+              .map((_, i) => i + 2)
+              .reverse(),
+            'Jobs are sorted by modify index'
+          );
+
+          // Simulate one of the on-page jobs getting its modify-index bumped. It should remain in place.
+          let existingJobToUpdate = server.db.jobs.findBy(
+            (job) => job.modifyIndex === 5
+          );
+          server.db.jobs.update(existingJobToUpdate.id, { modifyIndex: 12 });
+          await controller.watchJobIDs.perform(currentParams, 0);
+          let updatedOnPageJob = assert.async();
+
+          setTimeout(async () => {
+            rows = document.querySelectorAll('.job-row');
+            modifyIndexes = Array.from(rows).map((row) =>
+              parseInt(row.getAttribute('data-test-modify-index'))
+            );
+            assert.deepEqual(
+              modifyIndexes,
+              [12, 11, 10, 9, 8, 7, 6, 4, 3, 2],
+              'Jobs are sorted by modify index, on-page job remains in-place, and off-page pending'
```

Comment:
woop, same misleading message here too.  I think the actual logic is right for the live-updates-enabled case we're in here, just not the comment and this assertion message.

Reply:
Thanks for the catch, and yeah validated this is the behaviour from the actual endpoint too.

That said: I have eyes on doing some in-browser post-fetch magic at some point. Remembering the order the jobs were requested, and keeping them in that order regardless of the subsequent return. That way the user wouldn't see things dancing up to the top of the page if they had jostling disabled.

But that's for later if ever!